### PR TITLE
New Tier Three Product

### PR DESCRIPTION
**Pull request by rupertbates**
## What are you doing in this PR?
This PR adds the new Tier Three product into the code base as a separate product rather than an additional rate plan in Supporter Plus. Reasons for this are explained in [this document](https://docs.google.com/document/d/1NOata7_3YNsZ97bd312QQ9MYJrmko55CaO4D77ExUhI/edit?usp=sharing)

## TODOs (for follow up PRs)
- [ ] Finish off email integration, this is just a placeholder currently 
- [ ] Any extra validation needed beyond Guardian Weekly validation

### CODE REVIEW

File: support-frontend/app/controllers/CreateSubscriptionController.scala [hunk 1]
```diff
@@ -339,8 +339,12 @@ class CreateSubscriptionController(
         )
       case _: SupporterPlus =>
         List(
-          "gu_digital_subscriber" -> true.toString,
-          // "gu_supporter_plus" -> true.toString, // TODO: add this and remove the digisub one when the CMP cookie list has been updated
+          "gu_supporter_plus" -> true.toString,
```
Comment (rupertbates) [inline support-frontend/app/controllers/CreateSubscriptionController.scala:342]: This is in the cmp now

File: support-frontend/app/utils/CheckoutValidationRules.scala [hunk 2]
```diff
@@ -102,6 +102,7 @@ object CheckoutValidationRules {
       case d: DigitalPack => DigitalPackValidation.passes(createSupportWorkersRequest, d)
       case p: Paper => PaperValidation.passes(createSupportWorkersRequest, p.fulfilmentOptions)
       case _: GuardianWeekly => GuardianWeeklyValidation.passes(createSupportWorkersRequest)
+      case _: TierThree => GuardianWeeklyValidation.passes(createSupportWorkersRequest) // TODO: is this right?
```
Comment (jamesgorrie) [inline support-frontend/app/utils/CheckoutValidationRules.scala:105]: I think this is - but might be worth leaving a note why it's GWValidation (basically it's the same fields).

File: support-models/src/main/scala/com/gu/support/catalog/Product.scala [hunk 3]
```diff
@@ -42,6 +42,37 @@ sealed trait Product {
...
+      id: String,
+      billingPeriod: BillingPeriod,
+      fulfilmentOptions: FulfilmentOptions = NoFulfilmentOptions,
+  ) =
+    ProductRatePlan(
+      id,
+      billingPeriod,
+      fulfilmentOptions,
+      NoProductOptions,
+      s"Supporter Plus ${billingPeriod.getClass.getSimpleName}",
```
Comment (jamesgorrie) [inline support-models/src/main/scala/com/gu/support/catalog/Product.scala:56]: Is this right?

File: support-workers/src/main/scala/com/gu/emailservices/ThreeTierEmailFields.scala [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import com.gu.support.workers.states.SendThankYouEmailState.SendThankYouEmailTierThreeState
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ThreeTierEmailFields(
+    paperFieldsGenerator: PaperFieldsGenerator,
+    touchPointEnvironment: TouchPointEnvironment,
+) {
+  def build(state: SendThankYouEmailTierThreeState)(implicit ec: ExecutionContext): Future[EmailFields] = {
+    // TODO: don't really know what we need here yet
```
Comment (jamesgorrie) [inline support-workers/src/main/scala/com/gu/emailservices/ThreeTierEmailFields.scala:14]: Nice to know where this is now!

File: support-workers/src/main/scala/com/gu/support/workers/lambdas/FailureHandler.scala [hunk 5]
```diff
@@ -42,6 +42,10 @@ class FailureHandler(emailService: EmailService) extends Handler[FailureHandlerS
         FailedEmailFields.contribution(email = state.user.primaryEmailAddress, IdentityUserId(state.user.id))
       case _: SupporterPlus =>
         FailedEmailFields.supporterPlus(email = state.user.primaryEmailAddress, IdentityUserId(state.user.id))
+
+      case _: TierThree =>
+        // TODO: Do we need a new one here?
+        FailedEmailFields.supporterPlus(email = state.user.primaryEmailAddress, IdentityUserId(state.user.id))
```
Comment (jamesgorrie) [inline support-workers/src/main/scala/com/gu/support/workers/lambdas/FailureHandler.scala:48]: Is this going to be closer to GW given that we might want to pass delivery details in the email. 

This is a consideration that we can look at while looking at the email.

**Review by jamesgorrie — APPROVED** [review]
This looks good - there's a question that looks like a copy/paste problem.

I was also wondering if we should have the email more aligned to delivery products i.e. GW whereas we're doing it more like S+ - but this is something we'll probably look at when doing the email.

File: support-models/src/main/scala/com/gu/support/catalog/Product.scala (hunk 3 above)
Comment (rupertbates) [reply support-models/src/main/scala/com/gu/support/catalog/Product.scala:56]: Nope, thanks.

**Pull request merged**